homeassistant/components/usb/__init__.py

```diff
@@ -226,6 +254,8 @@ def __init__(
         self._request_callbacks: list[CALLBACK_TYPE] = []
         self.initial_scan_done = False
         self._initial_scan_callbacks: list[CALLBACK_TYPE] = []
+        self._port_event_callbacks: list[PORT_EVENT_CALLBACK_TYPE] = []
```

Comment:
```suggestion
        self._port_event_callbacks: set[PORT_EVENT_CALLBACK_TYPE] = set()
```

Its faster to lookup a callback in a set